src/QLContext.actor.cpp

```diff
@@ -530,6 +543,14 @@ struct SimpleIndexPlugin : IndexPlugin, ReferenceCounted<SimpleIndexPlugin>, Fas
 				// fprintf(stderr, "New value: %s\n", printable(StringRef(v.encode_key_part())).c_str());
 				DataKey new_key(self->indexPath);
 				new_key.append(v.encode_key_part()).append(documentPath[documentPath.size() - 1]);
+				if (new_key.byteSize() > DocLayerConstants::INDEX_KEY_LENGTH_LIMIT) {
+					TraceEvent(SevError, "SimpleIndexKeyTooLarge")
+					    .detail("Offending Key size", new_key.byteSize())
+					    .detail("Index name", DataValue::decode_key_part(
```

Comment:
Even though it's not directly related, one improvement could be store `IndexInfo` in `IndexPlugin` and avoid storing `indexPath`, `multikey` and `isUniqueIndex` separately. That way you can also read index name from IndexInfo in string format, rather doing decoding from index key.